pkg/runtime/runtime.go

```diff
@@ -676,30 +676,39 @@ func (a *DaprRuntime) populateSecretsConfiguration() {
 	}
 }
 
-func (a *DaprRuntime) buildHTTPPipelineForSpec(spec config.PipelineSpec, targetPipeline string) (httpMiddleware.Pipeline, error) {
-	var handlers []httpMiddleware.Middleware
-
+func (a *DaprRuntime) buildHTTPPipelineForSpec(spec config.PipelineSpec, targetPipeline string) (pipeline httpMiddleware.Pipeline, err error) {
 	if a.globalConfig != nil {
+		pipeline.Handlers = make([]func(next nethttp.Handler) nethttp.Handler, 0, len(spec.Handlers))
 		for i := 0; i < len(spec.Handlers); i++ {
 			middlewareSpec := spec.Handlers[i]
 			component, exists := a.getComponent(middlewareSpec.Type, middlewareSpec.Name)
 			if !exists {
-				err := fmt.Errorf(
-					"couldn't find middleware component with name %s and type %s/%s",
-					middlewareSpec.Name, middlewareSpec.Type, middlewareSpec.Version,
-				)
-				return httpMiddleware.Pipeline{}, err
+				log.Warn("error processing middleware component, daprd process will exit gracefully")
```

Comment:
Is there a valid scenario that sidecar should continue if cannot find the component? For example, can there be an app that is trying to use a middleware that might not be configured yet due to multiple versions of the app running and the component is not created yet.

Reply:
I was wondering about whether there could be a valid scenario, but eventually I thought I shouldn't allow this because it could lead to unexpected situations. My thinking was that a situation where the Configuration references a middleware that doesn't exist is more likely than not a mistake and allowing Dapr to init anyways could cause errors to go unnoticed. For multiple versions, one could use multiple Configuration resources.

However i am happy to change this if you prefer.

Reply:
Can we not change the behavior unless it is part of the hotix?

Reply:
Ok, I will change that behavior tomorrow morning

Reply:
Done